pkg/limayaml/limayaml.go

```diff
@@ -229,6 +230,12 @@ type Provision struct {
 	Script                          string             `yaml:"script" json:"script"`
 	File                            *LocatorWithDigest `yaml:"file,omitempty" json:"file,omitempty" jsonschema:"nullable"`
 	Playbook                        string             `yaml:"playbook,omitempty" json:"playbook,omitempty"`
+	// The following fields must be nil unless Mode is ProvisionModeData
```

Comment:
Can we just use an embedded struct?
```go
type ProvisionData struct{
  Content *string `...`
  // ...
}
type Provision struct {
  Mode ProvisionMode `...`
  // ...
  *ProvisionData
}
```

Reply:
Sure, as long as it is transparent to YAML unmarshaling (I just don't know; will test). I wouldn't want to have an extra settings level for them in `lima.yaml`.

Reply:
This seems to break "strict" YAML mode:

```
time="2025-03-11T23:03:06-07:00" level=warning msg="Non-strict YAML detected; please check for typos" comment="main file \"lima.yaml\"" error="[1:26] unknown field \"path\"\n>  1 | provision: [{mode: data, path: /tmp, content: hello}]\n                                ^\n   2 | images: [{location: /}]"
```

It looks like go-yaml has a tag to flatten promoted fields, so I'm going to use that:

```go
	*ProvisionData `yaml:",inline"` // Flatten fields for "strict" YAML mode
```

Not sure about making the embedded ProvisionData a pointer; it will just require additional checks, and validation already checks each individual field. But I'll try.